proposed/2022/DotnetListPackageMachineReadableJsonOutput.md

```diff
@@ -772,6 +1020,240 @@ Outputs json for format version 1, if it's not specified then latest version'll
 }
 ```
 
+### Error/Warning handling
+
+In case of an `error` it would be written into `problems` section and return non-0 error exit code if there is any `error` to indicate there had been some error to help scripting use case. Note: `Warnings` in `problems` section are not considered as `error`.
+
+| # | Scenario | Behavior in dotnet cli | Exit code
+|---|---|---|---|
+| 1 | Missing .net sdk/runtime installion| No json output, show missing .net sdk/runtime installation error in console | ?
+| 1 | Runtime error | No json output, because it needs all the data for json output, normal error stack output in console| 1
+| 1 | Parameter/options typo error | May defaults back to console output instead of json output, for example: `dotnet list package -include-transitive --format json` (here `-include-transitive` is missing another `-` in front), because that option/parameter validation for `-include-transitive` happen way before `list package` code execution | 1
+| 1 | Passing unsupported targetframe | `dotnet list package -framework net9.0` wouldn't fail even though `net9.0` is not valid framework at this moment, it would simply return empty result.  | 0
+| 1 | Asset file missing | Error would be in problems section of json, see below. | 1
+| 1 | Non-existing csproj file path | Error would be in problems section of json, see below. | 1
+| 1 | Package.config project | Error would be in problems section of json, see below. | 1
+| 1 | Incompatible option combination | Warning would be in problems section of json, see below. | 0
+| 1 | Use of not secure http source | Warning would be in problems section of json, see below. | 0
+| 1 | Unsupported output format | Defaults back to console output, for example: `dotnet list package --format yaml` | 1
+| 1 | Unsupported output version | if it's for json output then no json output, error out unsupported output version , see below | 1
+| 1 | Unsupported output version | if it's for console output then it's ignored, versioning is not considered for console output | 0
```

Comment:
@NuGet/nuget-client 
Should we warn if customer passes unsupported output version for console output? Technically console output can just ignore it.

Reply:
imo we should error for unsupported versions. 

The customers wants something very specific and they can't get it with that version of tooling so they should know that.

Reply:
Sure, we need to error out, updated the spec doc, I'll make change for `console output error out` implementation with https://github.com/NuGet/Home/issues/12167